### DISCUSSION TITLE
Imply HOMEBREW_NO_INSTALL_FROM_API=1 when developer mode is enabled

### DISCUSSION THREAD

**Discussion by Omoeba** [Homebrew/discussions#4536] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.19-24-ge96fd24
ORIGIN: https://github.com/Homebrew/brew
HEAD: e96fd24a7328a4913de2f7c56fd2c97d9aaa4572
Last commit: 35 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: abee8dab78f71a93b46a111de8bd7bd3138f9216
Core tap last commit: 4 hours ago
Core tap branch: master
Core tap JSON: 22 May 09:49 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.iKaXQB6ZTk/org.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_EVAL_ALL: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo openresty/brew) checkout master
```

### Description of issue

This is my first time using discussions so apologies if this is the wrong place or the label is incorrect. After seeing issues like https://github.com/Homebrew/brew/issues/15342, I think it may be a good idea to automatically tap `homebrew/core` and `homebrew/cask` when developer mode is enabled either by `brew developer on` or automatically when a developer command such as `brew bump-cask-pr` is run for the first time in addition to disabling the API. It'll make the first-time contributor experience much more straightforward and reduce bug reports due to confusing behaviors and error messages. A possible alternative is automatically tapping the core taps and updating them when a developer command such as `brew bump-cask-pr` is run but leaving the API enabled for general usage.

**Comment by carlocab** [comment, marked as answer]
You can file this as a feature request at https://github.com/Homebrew/brew/issues/new/choose.

**Comment by MikeMcQuaid** [comment]
> A possible alternative is automatically tapping the core taps and updating them when a developer command such as `brew bump-cask-pr` is run but leaving the API enabled for general usage.

This sounds better to me 👍🏻
